src/vault_snapshot/main.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import logging
+import os
+
+import context
+
+from lib.logging import setup_logging
+from lib.validations import hostname_agrees_with_artsy_environment
+
+from snap import take_snapshot
+
+
+def parse_args():
+    """parse command line args"""
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "artsy_env",
+        choices=["staging", "production"],
+        help="the artsy environment of the Vault instance",
+    )
+    parser.add_argument(
+        "--loglevel",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        default="INFO",
+        help="log level",
+    )
+    parser.add_argument(
+        "--s3", action="store_true", help="indicates to save snapshot to s3"
+    )
+    return parser.parse_args()
+
+
+def parse_env():
+    """parse env vars"""
+    vault_host = os.environ.get("VAULT_HOST")
+    vault_port = os.environ.get("VAULT_PORT")
+    vault_role = os.environ.get("VAULT_ROLE")
+    s3_bucket = os.environ.get("VAULT_SNAPSHOT_S3_BUCKET", "")
+    s3_prefix = os.environ.get("VAULT_SNAPSHOT_S3_PREFIX", "dev")
+    # local dir to store snapshot
+    local_dir = os.environ.get("LOCAL_DIR", "/tmp/vault_snapshot")
+    return (local_dir, vault_host, vault_port, vault_role, s3_bucket, s3_prefix)
+
+
+def validate(artsy_env, vault_host, vault_port, s3, s3_bucket):
+    """validate config obtained from env and command line"""
+    if not (vault_host and vault_port):
+        raise Exception(
```

Comment:
This and the following exceptions could use the built-in `ValueError`. Use of generic `Exception` is generally to be avoided in favour of more specific classes.

Reply:
[agreed](https://github.com/artsy/opstools/pull/142/commits/ee273e3776afe7803ded380e7a419b8c8776de22).